src/power.jl

```diff
@@ -58,37 +58,95 @@ end
 
 ^(a::Taylor1{TaylorN{T}}, r::Rational) where {T<:NumberNotSeries} = a^(r.num/r.den)
 
+# in-place form of power_by_squaring
+# this method assumes `y`, `x` and `aux1` are of same order
+function power_by_squaring!(y::TaylorN{T}, x::TaylorN{T}, aux1::TaylorN{T}, p::Integer) where {T<:NumberNotSeries}
```

Comment:
Could an equivalent method be added for `Taylor1` inputs? Or are they not necessary?

Reply:
I think it would be useful, yes, I've added it, thanks!